libs/server/Storage/Session/ObjectStore/SortedSetOps.cs

```diff
@@ -927,5 +927,194 @@ private GarnetStatus SortedSetDifference<TObjectContext>(ReadOnlySpan<ArgSlice>
 
             return GarnetStatus.OK;
         }
+
+        /// <summary>
+        /// Computes the intersection of multiple sorted sets and returns the members and their scores.
+        /// </summary>
+        /// <typeparam name="TObjectContext"></typeparam>
+        /// <param name="keys"></param>
+        /// <param name="pairs"></param>
+        /// <param name="objectContext"></param>
+        /// <returns></returns>
+        public GarnetStatus SortedSetIntersection<TObjectContext>(ReadOnlySpan<ArgSlice> keys, double[] weights, SortedSetAggregateType aggregateType, ref TObjectContext objectContext, out Dictionary<byte[], double> pairs)
+            where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            pairs = default;
+
+            var statusOp = GET(keys[0].ToArray(), out var firstObj, ref objectContext);
+            if (statusOp == GarnetStatus.OK)
+            {
+                if (firstObj.garnetObject is not SortedSetObject firstSortedSet)
+                {
+                    return GarnetStatus.WRONGTYPE;
+                }
+
+                if (keys.Length == 1)
+                {
+                    pairs = firstSortedSet.Dictionary;
+                    return GarnetStatus.OK;
+                }
+
+                // Initialize result with first set
+                pairs = new Dictionary<byte[], double>(ByteArrayComparer.Instance);
+                foreach (var kvp in firstSortedSet.Dictionary)
+                {
+                    pairs[kvp.Key] = kvp.Value * (weights?.Length > 0 ? weights[0] : 1.0);
+                }
+
+                // Intersect with remaining sets
+                for (var i = 1; i < keys.Length; i++)
+                {
+                    statusOp = GET(keys[i].ToArray(), out var nextObj, ref objectContext);
+                    if (statusOp != GarnetStatus.OK)
+                    {
+                        pairs = default;
+                        return statusOp;
+                    }
+
+                    if (nextObj.garnetObject is not SortedSetObject nextSortedSet)
+                    {
+                        pairs = default;
+                        return GarnetStatus.WRONGTYPE;
+                    }
+
+                    foreach (var kvp in pairs)
+                    {
+                        if (!nextSortedSet.Dictionary.TryGetValue(kvp.Key, out var score))
+                        {
+                            pairs.Remove(kvp.Key);
+                        }
```

Comment:
nit: `continue;`

Reply:
I'm not sure I understood your intent. I changed it similar to Early Return Pattern, let me know if I am wrong